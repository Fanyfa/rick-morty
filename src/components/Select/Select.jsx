import { array, func, string } from 'prop-types';
import React, { Component } from 'react';
import styles from './styles.module.scss';

class Select extends Component {
  static propTypes = {
    className: string,
    categories: array,
    onChange: func.isRequired,
  };

  handleOnChange = event => {
    const value = event.currentTarget.value;
    this.props.onChange(value);
  };

  render() {
    const { className, categories } = this.props;
    return (
      <select className={`${styles.select} ${className}`} onChange={this.handleOnChange}>
        <option value="">All</option>
        {categories.map((category, index) => {
          return (
            <option key={index} value={category}>
              {category}
            </option>
          );
        })}
      </select>
    );
  }
}

export default Select;
