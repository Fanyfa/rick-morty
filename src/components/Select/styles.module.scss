@import '~styles/colors';

.select {
  border-radius: 5px;
  border: 1px solid $color3;
  outline: none;
}
