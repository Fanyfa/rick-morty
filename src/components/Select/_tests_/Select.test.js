import Select from 'components/Select/';
import React from 'react';
import renderer from 'react-test-renderer';

const CLASSNAME = 'Lorem Inpsum';
const ONCHANGE = () => {};
const CATEGORIES = ['Alive', 'unknown', 'Dead'];

describe('<Select>', () => {
  it('renders', () => {
    const tree = renderer.create(<Select categories={CATEGORIES} onChange={ONCHANGE} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {classname}', () => {
    const tree = renderer
      .create(<Select categories={CATEGORIES} className={CLASSNAME} onChange={ONCHANGE} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {categories}', () => {
    const tree = renderer.create(<Select categories={CATEGORIES} onChange={ONCHANGE} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
