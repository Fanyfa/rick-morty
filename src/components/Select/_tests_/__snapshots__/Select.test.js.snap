// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Select> renders 1`] = `
<select
  className="select undefined"
  onChange={[Function]}
>
  <option
    value=""
  >
    All
  </option>
  <option
    value="Alive"
  >
    Alive
  </option>
  <option
    value="unknown"
  >
    unknown
  </option>
  <option
    value="Dead"
  >
    Dead
  </option>
</select>
`;

exports[`<Select> when {categories} 1`] = `
<select
  className="select undefined"
  onChange={[Function]}
>
  <option
    value=""
  >
    All
  </option>
  <option
    value="Alive"
  >
    Alive
  </option>
  <option
    value="unknown"
  >
    unknown
  </option>
  <option
    value="Dead"
  >
    Dead
  </option>
</select>
`;

exports[`<Select> when {classname} 1`] = `
<select
  className="select Lorem Inpsum"
  onChange={[Function]}
>
  <option
    value=""
  >
    All
  </option>
  <option
    value="Alive"
  >
    Alive
  </option>
  <option
    value="unknown"
  >
    unknown
  </option>
  <option
    value="Dead"
  >
    Dead
  </option>
</select>
`;
